src/Microsoft.OData.Core/JsonLight/ODataJsonLightServiceDocumentSerializer.cs

```diff
@@ -132,5 +133,131 @@ private void WriteServiceDocumentElement(ODataServiceDocumentElement serviceDocu
             // "}"
             this.JsonWriter.EndObjectScope();
         }
+
+        /// <summary>
+        /// Asynchronously writes a service document in JsonLight format.
+        /// </summary>
+        /// <param name="serviceDocument">The service document to write.</param>
+        internal Task WriteServiceDocumentAsync(ODataServiceDocument serviceDocument)
+        {
+            Debug.Assert(serviceDocument != null, "serviceDocument != null");
+
+            return this.WriteTopLevelPayloadAsync(
+                async () =>
+                {
+                    // "{"
+                    await this.AsynchronousJsonWriter.StartObjectScopeAsync()
+                        .ConfigureAwait(false);
+
+                    // "@odata.context":...
+                    await this.WriteContextUriPropertyAsync(ODataPayloadKind.ServiceDocument)
+                        .ConfigureAwait(false);
+
+                    // "value":
+                    await this.AsynchronousJsonWriter.WriteValuePropertyNameAsync()
+                        .ConfigureAwait(false);
+
+                    // "["
+                    await this.AsynchronousJsonWriter.StartArrayScopeAsync()
+                        .ConfigureAwait(false);
+
+                    if (serviceDocument.EntitySets != null)
+                    {
+                        foreach (ODataEntitySetInfo collectionInfo in serviceDocument.EntitySets)
+                        {
+                            await this.WriteServiceDocumentElementAsync(collectionInfo, JsonLightConstants.ServiceDocumentEntitySetKindName)
+                                .ConfigureAwait(false);
+                        }
+                    }
+
+                    if (serviceDocument.Singletons != null)
+                    {
+                        foreach (ODataSingletonInfo singletonInfo in serviceDocument.Singletons)
+                        {
+                            await this.WriteServiceDocumentElementAsync(singletonInfo, JsonLightConstants.ServiceDocumentSingletonKindName)
+                                .ConfigureAwait(false);
+                        }
+                    }
+
+                    HashSet<string> functionImportsWritten = new HashSet<string>(StringComparer.Ordinal);
+
+                    if (serviceDocument.FunctionImports != null)

```

Comment:
Does this include all operation imports, both function and action imports?

Reply:
Yes. The more inclusive name would have been `OperationImports`

Reply:
@gathogojr since `FunctionImports` is a Public property, we should create an issue to fix this in the next breaking change release.